tools/istio-iptables/pkg/dependencies/implementation.go

```diff
@@ -90,20 +93,43 @@ func (r *RealDependencies) execute(cmd string, ignoreErrors bool, args ...string
 	return externalCommand.Run()
 }
 
-func (r *RealDependencies) executeXTables(cmd string, ignoreErrors bool, args ...string) error {
+func (r *RealDependencies) executeXTables(cmd string, ignoreErrors bool, args ...string) (err error) {
 	if r.CNIMode {
 		originalCmd := cmd
 		cmd = constants.NSENTER
 		args = append([]string{fmt.Sprintf("--net=%v", r.NetworkNamespace), "--", originalCmd}, args...)
 	}
 	log.Infof("Running command: %s %s", cmd, strings.Join(args, " "))
-	externalCommand := exec.Command(cmd, args...)
-	stdout := &bytes.Buffer{}
-	stderr := &bytes.Buffer{}
-	externalCommand.Stdout = stdout
-	externalCommand.Stderr = stderr
 
-	err := externalCommand.Run()
+	var stdout, stderr *bytes.Buffer
+	b := backoff.NewExponentialBackOff()
+	b.InitialInterval = 100 * time.Millisecond
+	b.MaxInterval = 1 * time.Second
+	b.Reset()
+	for i := 0; i < 10; i++ {
+		externalCommand := exec.Command(cmd, args...)
+		stdout = &bytes.Buffer{}
+		stderr = &bytes.Buffer{}
+		externalCommand.Stdout = stdout
+		externalCommand.Stderr = stderr
+		err = externalCommand.Run()
+		exitCode, ok := exitCode(err)
+		if !ok {
+			// cannot get exit code. consider this as non-retriable.
+			break
+		}
+
+		if !isXTablesLockError(stderr, exitCode) {
+			// Command succeeded, or failed not because of xtables lock.
+			break
+		}
+
+		// If command failed because xtables was locked, try the command again.
+		// Note we retry invoking iptables command explicitly instead of using the `-w` option of iptables,
+		// because as of iptables 1.6.x (version shipped with bionic), iptables-restore does not support `-w`.
+		time.Sleep(b.NextBackOff())
```

Comment:
on the last re-try,  this code will sleep and then exit the loop.

I think `backoff.Retry` or ``backoff.RetryWithTimeout` will do the right thing.

Reply:
Updated to use backoff.Retry, with max elapse time as 10s and max interval 2s.